### PR TITLE
pybind/mgr/restful: improve cert handling; work with vstart

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by tchaikov** [comment]
retest this please. see #15406

File: src/pybind/mgr/restful/module.py [hunk 1]
```diff
@@ -244,18 +245,43 @@ def _serve(self):
...
-        pkey = self.get_config_json("pkey") or '/etc/ceph/ceph-mgr-restful.key'
+        server_addr = self.get_config('server_addr') or '127.0.0.1'
+        server_port = int(self.get_config('server_port') or '8003')
+        self.log.info('server_addr: %s server_port: %d',
+                      server_addr, server_port)
+
+        cert_fname = None
+        pkey_fname = None
+
+        cert = self.get_config("cert")
```
Comment (b-ranto) [inline src/pybind/mgr/restful/module.py]: This should be get_config_json, not get_config. get_config returns the cluster config as in /config/cluster, not the config-keys. (the same should go for other get_config calls).

File: src/pybind/mgr/restful/module.py [hunk 2]
```diff
@@ -244,18 +245,43 @@ def _serve(self):
...
 
-        cert = self.get_config_json("cert") or '/etc/ceph/ceph-mgr-restful.crt'
-        pkey = self.get_config_json("pkey") or '/etc/ceph/ceph-mgr-restful.key'
+        server_addr = self.get_config('server_addr') or '127.0.0.1'
+        server_port = int(self.get_config('server_port') or '8003')
+        self.log.info('server_addr: %s server_port: %d',
+                      server_addr, server_port)
+
+        cert_fname = None
+        pkey_fname = None
```
Comment (b-ranto) [inline src/pybind/mgr/restful/module.py]: There is no need to initialize variables like this in python, they will be overwritten (re-initialized) by next assignment anyway (that holds even for if/else like you use later in the code). The same goes for cert_tmp and pkey_tmp.

Comment (liewegas) [inline src/pybind/mgr/restful/module.py]: ah, fixed

File: src/pybind/mgr/restful/module.py (hunk 1 above)
Comment (liewegas) [inline src/pybind/mgr/restful/module.py]: i'm not sure what you mean. see the config-key sets in vstart.sh; this is pulling those same values (and working).  same thing that dashboard is doing...

Comment (b-ranto) [inline src/pybind/mgr/restful/module.py]: Ah, sorry, I have misread that. I thought it was self.get('config') which is a completely different thing. I suppose this should work. The only question is whether we want to do the json serialization or not but I guess for the keys, it does make sense not to serialize.

**Comment by liewegas** [comment]
@b-ranto can you take another quick look at the test change too?

**Comment by b-ranto** [comment]
lgtm as well

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
@b-ranto @jcsp: several changes

- changed commands to have 'restful' prefix
- added get_mgr_id
- added get_config_prefix
- get_localized_config helper in restful
- used 'crt' and 'key' names instead of 'cert' and 'pkey' (to match file extensions)

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-05_22:19:51-rados-wip-sage-testing-distro-basic-smithi/

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-05_22:19:51-rados-wip-sage-testing-distro-basic-smithi/ looks ok, but make check needs retset

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Pull request merged**